pkg/provider/gitlab/gitlab.go

```diff
@@ -155,10 +156,28 @@ func (g *Gitlab) NewClient(ctx context.Context, store esv1beta1.GenericStore, ku
 	return g, nil
 }
 
-// Empty GetAllSecrets.
+// GetAllSecrets syncs all gitlab project variables into a single Kubernetes Secret.
 func (g *Gitlab) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
-	// TO be implemented
-	return nil, fmt.Errorf("GetAllSecrets not implemented")
+	if utils.IsNil(g.client) {
+		return nil, fmt.Errorf(errUninitializedGitlabProvider)
+	}
+
+	allData, _, err := g.client.ListVariables(g.projectID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	checkName := ref.Name != nil && len(ref.Name.RegExp) > 0
```

Comment:
If Gitlab does not support find by tags, having a name should be mandatory, and should properly error out to the user as a missing parameter. the current logic would just sync a `nil` secret. (also, probably this check should be made before `ListVariables` 😄

Reply:
Thanks for the hint.